apps/google-analytics-4/lambda/src/lib/verify-signed-request-middleware.ts

```diff
@@ -0,0 +1,79 @@
+import * as NodeAppsToolkit from '@contentful/node-apps-toolkit';
+import {
+  CanonicalRequest,
+  ContentfulContextHeader,
+  ContentfulHeader,
+} from '@contentful/node-apps-toolkit/lib/requests/typings';
+import express from 'express';
+import { InvalidSignature } from '../lib/errors/invalid-signature';
+import { UnableToVerifyRequest } from '../lib/errors/unable-to-verify-request';
+import { IncomingHttpHeaders } from 'http';
+
+export const verifySignedRequestMiddleware = (
+  req: express.Request,
+  _res: express.Response,
+  next: express.NextFunction
+) => {
+  const signingSecret = (process.env.SIGNING_SECRET || '').trim();
+  const canonicalReq = makeCanonicalReq(req);
+  let isValidReq = false;
+
+  try {
+    isValidReq = NodeAppsToolkit.verifyRequest(signingSecret, canonicalReq, 60); // 60 second TTL
+  } catch (e) {
+    console.error(e);
+    throw new UnableToVerifyRequest('Unable to verify request', {
```

Comment:
Don't you call `next(error)`? That will then be caught in the error handler. But I am not an express expert.

Reply:
oh yes, let me try that. i'm very new to express and I was just tying this into the custom error handling middleware added by @jsdalton in a likely very ignorant way :) – thanks for the note!

Reply:
I think that should still be possible with `next(new UnableToVerifyRequest(...))`. But maybe both works and both is valid.

Reply:
oh! nifty! I'll give that a try shortly :) 

Thanks @andipaetzold! I always like to learn the right way, when I'm doing something stupid/implementing an anti-pattern like this. Thanks for the time & energy you're spending to level me up 😄